cpp/src/jit/row_ir.cpp

```diff
@@ -100,14 +114,33 @@ std::string set_output::generate_code(instance_context& ctx,
   }
 }
 
-/// @brief returns true if the operator depends on the nullability of its operands
-static bool is_nullness_dependent_operator(cudf::ast::ast_operator op)
+operation::operation(opcode op, std::unique_ptr<node>* move_begin, std::unique_ptr<node>* move_end)
+  : id_(), op_(op), operands_(), type_()
+{
+  std::move(move_begin, move_end, std::back_inserter(operands_));
+  CUDF_EXPECTS(static_cast<size_type>(operands_.size()) == ast::detail::ast_operator_arity(op),
+               "Invalid number of arguments for operator.",
+               std::invalid_argument);
+  CUDF_EXPECTS(
+    operands_.size() > 0, "Operator must have at least one operand", std::invalid_argument);
+}
```

Comment:
I believe `ast::detail::ast_operator_arity(op)` is always > 0 (right?). If so, we can just remove this check.